core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java

```diff
@@ -1184,4 +1186,15 @@ public static String getQuerySetsOfGenes() {
         String fileName = portalProperties.getProperty(SETSOFGENES_LOCATION, null);
         return readFile(fileName);
     }
+
+    public static String getWholeSlideViewerToken()
+    {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        
+        if(authentication != null && authentication.isAuthenticated()) {
```

Comment:
in most cases there will be no secret key, so you should put a check to only do this when a secret key exists.